### PR TITLE
CLOUDP-167798: Migrate Project IP Access List CLI stores to v2 sdk

### PR DESCRIPTION
**Pull request by swattyT**
## Proposed changes

_Jira ticket:_ CLOUDP-# [CLOUDP-167798](https://jira.mongodb.org/browse/CLOUDP-167798)
EVG Patch: https://spruce.mongodb.com/version/64246234c9ec443a1567af91/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/store/project_ip_access_lists.go
Comment (swattyT) [inline internal/store/project_ip_access_lists.go:41]: Delete e2e tests are still failing due to versioning error:
`{"detail":"Invalid accept header or version date.","error":406,"errorCode":"INVALID_VERSION_DATE","parameters":[],"reason":"Not Acceptable"}
Error: 406 Not Acceptable
`

Comment (wtrocki) [reply internal/store/project_ip_access_lists.go:41]: Working on fix.

Comment (swattyT) [reply internal/store/project_ip_access_lists.go:41]: no rush :)

File: internal/cli/atlas/accesslists/create.go
Comment (swattyT) [reply internal/cli/atlas/accesslists/create.go:73]: Just checked, direct access doesn't give a pointer either. The field is from the `GlobalOpts struct` in CLI.

Comment (wtrocki) [reply internal/cli/atlas/accesslists/create.go:73]: `&struct.Value` gives pointer in this case :)

File: go.sum
Comment (matt-condon) [inline go.sum:403]: Happy to merge these changes back to the vbranch provided we stick with the same sdk v2 version, so let's revert these changes and run a quick smoke test to validate before merging

**Review by matt-condon — APPROVED** [review]
LGTM once the sdk upgrade is reverted

**Pull request merged**
